src/back/glsl.rs

```diff
@@ -874,25 +911,47 @@ fn write_expression<'a, 'b>(
                 }
             };
 
-            //TODO: fix this
-            let sampler_constructor = format!(
-                "{}sampler{}{}{}({})",
-                match kind {
-                    ScalarKind::Sint => "i",
-                    ScalarKind::Uint => "u",
-                    ScalarKind::Float => "",
-                    _ => return Err(Error::Custom(String::from("Cannot build image of bools",))),
-                },
-                ImageDimension(dim),
-                if ms { "MS" } else { "" },
-                if arrayed { "Array" } else { "" },
-                image_expr,
-            );
-
-            let expr = if !ms {
-                format!("texture({},{})", sampler_constructor, coordinate_expr)
-            } else {
-                todo!()
+            let expr = match class {
+                ImageClass::Sampled | ImageClass::Multisampled => {
+                    let ms = match class {
+                        crate::ImageClass::Multisampled => true,
+                        _ => false,
+                    };
+
+                    //TODO: fix this
```

Comment:
is this `TODO` still needed?

Reply:
This part will be removed when we implement combined image samplers